### PR TITLE
Compiling debug with stack protection flags (-fstack-check -fstack-protector)

### LINKED ISSUE #1359: GCC stack protector
When adding the options 

    -fstack-check -fstack-protector

to the compiler, the broker doesn't compile.
These extra warnings might be very useful for the broker, we should use this (only as debugging tools)

**Effort: 3 man day**

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #1359 

Some steps has been done enabling -fstack-check -fstack-protector in debug mode but qStringFilters() still needs to be refactored (currently, this PR branch doesn't compile in debug mode).

### CODE REVIEW

File: src/lib/mongoBackend/MongoGlobal.cpp [hunk 1]
```diff
@@ -948,7 +949,7 @@ static void qStringFilters(const std::string& in, std::vector<BSONObj> &filters)
     }
     else
     {
-      printf("\nNo OP found\n");
+      //printf("\nNo OP found\n");
```
Comment (fgalan) [inline src/lib/mongoBackend/MongoGlobal.cpp]: Remove leftover

File: CMakeLists.txt [hunk 2]
```diff
@@ -80,6 +79,7 @@ if (${CMAKE_BUILD_TYPE} STREQUAL DEBUG)
 
     MESSAGE("cmake: DEBUG compilation")
     add_definitions(-DDEBUG -DDEBUG_$ENV{USER} -DDEBUG_${USER})
+    add_definitions(-fstack-check -fstack-protector)
```
Comment (fgalan) [inline CMakeLists.txt]: Suggestion for the comment:

"Note that stack protection is activated only in debug compilation. This is due that this directive makes the compiler introduce code at the begin/end of the functions, so has impact in performance (see <links where that is explained in more detail>)"

File: src/lib/mongoBackend/MongoGlobal.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: Fixed in 9165059

File: CMakeLists.txt (hunk 2 above)
Comment (kzangeli) [inline CMakeLists.txt]: Fixed in d8d8681

**Comment by fgalan** [comment]
LGTM

**Pull request merged**